test/whitequark/test_context_lambda.rb

```diff
@@ -0,0 +1,4 @@
+# typed: strict
+
+module A; ->(a = (return; 1)) {}; end
+class B; ->(a = (return; 1)) {}; end
```

Comment:
The whitequark PR also mentioned another case where the `module` was inside the lambda. I guess maybe that test doesn't make sense to add until Sorbet supports `_1` syntax?

Reply:
The case is currently accepted by Sorbet: https://sorbet.run/#%23%20typed%3A%20true%0A%0A-%3E(a)%20%7B%20module%20M%3B%20%22%22%3B%20end%20%7D

We need this one to be merged before #3344 so I can port the fix.